packages/devtools_server/lib/src/server.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:args/args.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+import 'package:shelf/shelf.dart' as shelf;
+import 'package:shelf/shelf_io.dart' as shelf;
+import 'package:shelf_static/shelf_static.dart';
+
+const argHelp = 'help';
+const argMachine = 'machine';
+const argPort = 'port';
+
+final argParser = new ArgParser()
+  ..addFlag(
+    argHelp,
+    negatable: false,
+    abbr: 'h',
+    help: 'Prints help output.',
+  )
+  ..addOption(
+    argPort,
+    defaultsTo: '9100',
+    abbr: 'p',
+    help: 'Port to serve DevTools on. '
+        'Pass 0 to automatically assign an available port.',
+  )
+  ..addFlag(
+    argMachine,
+    negatable: false,
+    abbr: 'm',
+    help: 'Sets output format to JSON for consumption in tools.',
+  );
+
+void serveDevTools(List<String> arguments) async {
```

Comment:
bonus points: make a second method that takes a bunch of named parameters instead of a list String arguments.